### PR TITLE
[MRG] Fixing scans and adding unit test when anonymizing.

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #351 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by jasmainak** [comment]
Thanks @adam2392 !

File: doc/whats_new.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ Bug
 - Fixed broken event timing broken by conversion to BV in :func:`write_raw_bids`, by `Alex Rockhill`_ (`#294 <https://github.com/mne-tools/mne-bids/pull/294>`_)
 - Support KIT related files .elp and .hsp BIDS conversion in :func:`write_raw_bids`, by `Fu-Te Wong`_ (`#323 <https://github.com/mne-tools/mne-bids/pull/323>`_)
 - Enforce that the name arg is a required field for :func:`mne_bids.make_dataset_description`, by `Teon Brooks`_ and `Stefan Appelhoff`_ (`#342 <https://github.com/mne-tools/mne-bids/pull/342>`)
+- Fix writing to scans.tsv file when anonymization is turned on, by `Adam Li`_ (`#352 <https://github.com/mne-tools/mne-bids/pull/352>`)
```
Comment (jasmainak) [inline doc/whats_new.rst]:
```suggestion
- Fix writing to scans.tsv file when anonymization is turned on, by `Adam Li`_ (`#352 <https://github.com/mne-tools/mne-bids/pull/352>_`)
```
oops looks like I merged too soon

Comment (jasmainak) [reply doc/whats_new.rst]: can you make another PR to fix it?

Comment (adam2392) [reply doc/whats_new.rst]: @jasmainak I'll fix it in my other PR. my bad!

**Pull request merged**